geomstats/geometry/trees.py

```diff
@@ -0,0 +1,485 @@
+""" Classes for trees and forests, splits and structures.
+
+An instance of class Wald (Tree or Forest) is an element of the Wald Space or BHV space,
+for example.
```

Comment:
BHV spaces as well? Are we going to use this class for the bhv or we are going to split it